### ISSUE TITLE
Issue generating ligand

### ISSUE THREAD

**Issue by paul-goldsmith** [paiardin/DockingPie#6]
Hi, I am running through the tutorial example using PDB: 1OL5 but keep getting an error when clicking "Generate Ligand".  All steps up to that point are fine.  I have configured and installed everything as in the User Guide.

Traceback (most recent call last):
  File "C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol_311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\docking_program_gui\tabs.py", line 2183, in generate_ligand_func
    generated_ligand = Generate_Object(self, dict = self.docking_programs_child_tabs.docking_programs.vina_ligands_dict,
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol_311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\Functions\general_docking_func.py", line 284, in __init__
    self.generate_receptor()
  File "C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol_311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\Functions\general_docking_func.py", line 306, in generate_receptor
    self.generate_checked_object()
  File "C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol_311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\Functions\general_docking_func.py", line 412, in generate_checked_object
    os.rename(old_name, new_name)
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'obj01.pdbqt' -> '01_obj01_Vina.pdbqt'

# DockingPie
- Version: 1.2.1
- Plugin path: C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol_311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main 
- Config directory: C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol_311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\config

# PyMOL
- Version: 2.6.0a0
- Path: C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol_311\pythonw.exe
- Qt: PyQt5
- Has Conda: False

# Python
- Version: 3.11.7 | packaged by Anaconda, Inc. | (main, Dec 15 2023, 18:05:47) [MSC v.1916 64 bit (AMD64)]
- Arch: 64
- Path: C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol_311\pythonw.exe

# Operating system
- Platform: win32
- Arch: 64

# Python libs
- PyQt5: 5.15.10
- Conda version: -
- Numpy version: 1.26.4
- Biopython version: 1.83

**Comment by SerenaRosi** [comment]
Please double check that the plugin is properly configured for all dependencies in the CONFIGURATION tab.

**Comment by paul-goldsmith** [comment]
They all appear to be OK, the status by each of them is:
RxDock is not available for this OS
Vina already installed
Smina is not available for this OS
ADFR already installed
Openbabel already installed
sPyRMSD already installed
sdsorter is not available for this OS

**Comment by paul-goldsmith** [comment]
I get the error for generating ligands in both Vina and ADFR tabs

**Comment by SerenaRosi** [comment]
We have recently become aware of unexpected issues reported by multiple users, which we believe may be stemming from recent updates to PyMOL and/or other external libraries. These issues were not encountered previously and are likely a result of incompatibilities introduced by these updates. As a result, we have temporarily placed DockingPie under maintenance mode while we work diligently to resolve these issues. Once these issues are resolved, we will promptly update our GitHub page to inform users of the fixes and any necessary instructions. Thank you for your understanding and cooperation during this period.

Best,
Serena

**Comment by paiardin** [comment]
We have just released an updated version of [DockingPie](https://github.com/paiardin/DockingPie/archive/refs/heads/main.zip), which should fix this issue.

**Comment by paul-goldsmith** [comment]
Hi Serena,
I'm afraid it isn't working for me still, this time there is an issue generating the receptor.  After selecting the receptor and clicking "Import" nothing gets imported into the generate receptor part of the window.  (Not really relevant but repeated clicking just cycles the receptor through different colours).  The text in the GUI is pasted below:

PyMOL>fetch 1ol5
TITLE     Structure of Aurora-A 122-403, phosphorylated on Thr287, Thr288 and bound to TPX2 1-43
 ExecutiveLoad-Detail: Detected mmCIF
 CmdLoad: ".\1ol5.cif" loaded as "1ol5".
 You clicked /1ol5/C/A/ADP`1388/C4
 Selector: selection "sele" defined with 27 atoms.
Traceback (most recent call last):
  File "C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol25conda_py311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\docking_program_gui\new_windows.py", line 104, in <lambda>
    self.main_button.clicked.connect(lambda a=None: self.submit_command())
                                                    ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol25conda_py311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\Functions\pymol_interactions.py", line 298, in import_selected_pymol_object
    self.import_selected_pymol_receptors()
  File "C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol25conda_py311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\Functions\pymol_interactions.py", line 329, in import_selected_pymol_receptors
    Load_Object.load_checked_structures(self = self,
  File "C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol25conda_py311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\Functions\pymol_interactions.py", line 767, in load_checked_structures
    if self.receptor:
       ^^^^^^^^^^^^^
AttributeError: 'Import_from_Pymol' object has no attribute 'receptor'

**Comment by SerenaRosi** [comment]
Hello, I am really sorry about that. The issue is now solved. 
Thank you for your patience and interest in DockingPie. 

Best, 
Serena

**Comment by paul-goldsmith** [comment]
Hi Serena,
Sorry to keep coming back to you but thank you for trying to fix this, I am very keen to use DockingPie!  
The receptor issue is fixed but I now still have the ligand generation issue that I had originally.  I saw there is more information in the terminal window compared to the PyMol GUI so I am pasting it all here.  It might be an issue with generating the pdbqt file - there is the error "ImportError: No module named MolKit".  All text pasted below.
Thanks,
Paul


PyMOL>fetch 1ol5
TITLE     Structure of Aurora-A 122-403, phosphorylated on Thr287, Thr288 and bound to TPX2 1-43
 ExecutiveLoad-Detail: Detected mmCIF
 CmdLoad: ".\1ol5.cif" loaded as "1ol5".
 You clicked /1ol5/C/A/ADP`1388/C4
 Selector: selection "sele" defined with 27 atoms.
C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol25conda_py311\Lib\site-packages\Bio\PDB\StructureBuilder.py:89: PDBConstructionWarning: WARNING: Chain A is discontinuous at line 2422.
  warnings.warn(
C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol25conda_py311\Lib\site-packages\Bio\PDB\StructureBuilder.py:89: PDBConstructionWarning: WARNING: Chain B is discontinuous at line 2572.
  warnings.warn(
Traceback (most recent call last):
  File "C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol25conda_py311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\config\prepare_receptor4.py", line 20, in <module>
    import AutoDockTools.MolKit as mk
ImportError: No module named MolKit
 Executive: object "temp_obj" created.
 Executive: object temp_obj created.
Traceback (most recent call last):
  File "C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol25conda_py311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\config\prepare_ligand4.py", line 18, in <module>
    import AutoDockTools.MolKit as mk
ImportError: No module named MolKit
Traceback (most recent call last):
  File "C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol25conda_py311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\docking_program_gui\tabs.py", line 2183, in generate_ligand_func
    generated_ligand = Generate_Object(self, dict = self.docking_programs_child_tabs.docking_programs.vina_ligands_dict,                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^  File "C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol25conda_py311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\Functions\general_docking_func.py", line 284, in __init__
    self.generate_receptor()
  File "C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol25conda_py311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\Functions\general_docking_func.py", line 306, in generate_receptor
    self.generate_checked_object()
  File "C:\Users\pgoldsmith\AppData\Local\anaconda3\envs\pymol25conda_py311\Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\Functions\general_docking_func.py", line 412, in generate_checked_object
    os.rename(old_name, new_name)
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'obj01.pdbqt' -> '01_obj01_Vina.pdbqt'

**Comment by SerenaRosi** [comment]
It looks like your issue might be related to the installation of Openbabel. To help you better, could you please provide some additional information? 

How did you install Openbabel?

Have you tried working with an Incentive version of PyMOL, or one of those reported as tested? (attached below)


PyMOL version | Operating system | PyMOL source
3.0.0 | Linux (Ubuntu 22.04.3 LTS), 64-bit | Incentive
3.0.0 | Windows (v.11), 64-bit | Incentive
3.0.0 | MacOS (High Sierra and above) | Incentive
2.6.0 | Linux (Ubuntu 22.04.3 LTS), 64-bit | Incentive
2.6.0 | Windows (v.11), 64-bit | Incentive
2.6.0 | MacOS (High Sierra and above) | Incentive
2.5.7 | Linux (Ubuntu 22.04.3 LTS), 64-bit | Incentive
2.5.7 | Windows (v.11), 64-bit | Incentive
2.5.7 | MacOS (High Sierra v.10.13.6) | Incentive
2.5.5 | Linux (Ubuntu 22.04.3 LTS), 64-bit | Incentive
2.5.5 | Windows (v.11), 64-bit | Incentive
2.5.5 | MacOS (High Sierra v.10.13.6) | Incentive
2.5.2 | Linux (Ubuntu 20.04.3 LTS), 64-bit | Incentive
2.5.1 | Linux (Ubuntu 21.04), 64-bit | Incentive (Conda package)
2.5.2 | Linux (Ubuntu 18.04.2 LTS), 64-bit | Incentive
2.5.0 | Linux (Ubuntu 21.04), 64-bit | Open source (Conda package)
2.5.0 | Linux (Ubuntu 20.04.3 LTS), 64-bit | Open source (Conda package)
2.4.1 | Windows (v.10 Home), 64-bit | Incentive
2.5.2 | Windows (v.10 Pro), 64-bit | Incentive
2.5.0 | Windows (v.10 Pro), 64-bit | Open source (Conda package)
2.5.2 | MacOS (High Sierra v.10.13.6) | Incentive
2.4.0 | MacOS (High Sierra v.10.13.6) | Open source (Conda package)
2.5.2 | MacOS (Monterey v.12.2.1) | Incentive
2.5.2 | MacOS (Big Sur v.11.6.5) | Incentive
2.4.0 | MacOS (Big Sur v.11.6.5) | Open source (Conda package)

**Comment by paul-goldsmith** [comment]
I am using open source pymol 2.50 (from conda) on Win10 Pro 64-bit using python 3.11.  Open Babel was installed through conda-forge as per the instructions in your PDF (along with the other required pakages).  
It is all setup in a new environment using Anaconda Navigator.

**Comment by paul-goldsmith** [comment]
I am not an expert in python at all so this might be completely wrong but think I've narrowed the issue down to the program not knowing where to look to load the MolKit module in the prepare_ligand.py file.  

If I open that python file in an editor (Spyder) and just try and run the opening line 
import Autodock.MolKit as mk
I get the error:
ImportError: No module named MolKit

I then installed the AutoDockTools into my environment (which put it into site-packages) and found that if I then just run 
import MolKit as mk
the module loaded with no error as the Molkit folder is within site-packages.

So, I think for the installation into the pmg_tk directory, the code is not searching the folder structure you have set up with Lib\site-packages\pmg_tk\startup\DockingPie1\lib\docking_program_main\config\AutoDockTools\MolKit.

Hopefully that makes some sense.
